fix: decorations working for long notes, improved task note decorations

SeriousBug opened the pull request:
- Thanks to an optimization, all decorations (like link highlighting) are now displayed even in very long notes.
- Fixed a bug with decorations not appearing in Native Workspaces.
- Task note decorations are colored with the same color as the link to make them easier to distinguish from regular text.
- Task notes say `priority:` instead of `prio:` for priorities.

![](https://i.imgur.com/rhCykfd.png)

kevinslin commented on packages/common-all/src/types/workspace.ts:
```diff
@@ -356,13 +356,7 @@ export type DendronConfig = {
-   * Defaults to 4096 characters, which is about 4 KiB.
-   */
+  /** @deprecated No longer relevant, all decorations are now displayed for long notes. */
   maxNoteDecoratedLength?: number;
```
if we're no longer using, does it make sense to just remove altogether?

kevinslin commented on packages/plugin-core/src/features/windowDecorations.ts:
```diff
@@ -126,114 +121,108 @@ export function delayedUpdateDecorations(
+          allWarnings.push(warnMissingFrontmatter(activeEditor.document));
+        } else {
+          allWarnings.push(
+            ...warnBadFrontmatterContents(activeEditor.document, frontmatter)
```
can we rename this method? `warnBadFrontmatterContents` makes me think that is bad frontmatter. `checkAndWarnIfBadFrontmatterContents` would be more descriptive

kevinslin reviewed (changes requested):
Found an edge case where if I start from a task note and then go to a note that links to it, the decorators show up 2x: https://www.loom.com/share/f04e23bb082642f98221aa912ffba1d2?from_recorder=1&focus_title=1

Repro by going to [[Prepare Report|task.2021.10.26.020813.prepare-report]] in test-workspace

SeriousBug commented:
> Found an edge case

Oops! Thought I had fixed this with the `mergeRanges` function but looks like I used it in the wrong order :)

Fixed the bug and applied other feedback now.
